tests/test_setup.py

```diff
@@ -58,3 +66,47 @@ def test_more_metadata(self) -> None:
         assert len(long_description) > 7
         assert long_description[0].strip() != ""
         assert long_description[-1].strip() != ""
+
+    def test_build_extension(self) -> None:
+        # Do we handle all expected errors?
+        for ext_error in setup.ext_errors:
+            with pytest.raises(setup.BuildFailed):
+                setup.build_ext.build_extension = self.raise_error(ext_error)
```

Comment:
These tests are mostly checking that we catch certain errors.  I'd be more interested in tests that demonstrated actual failure scenarios and that we handled them well, but that seems harder to do.  I'm not sure we need these tests.

Reply:
Agreed, closing. If I figure out a way to construct actual failure scenarios I'll propose another PR with realistic tests. Thanks for taking a look!